### ISSUE TITLE
Question: Vite (Vue3) with electron - Best approach to encode?

### ISSUE THREAD

**Issue by cliqer** [OsamaAbbas/bytenode#69]
Hi Osama and thank you for this amazing project.

I am using [vite](https://github.com/vitejs/vite) which creates a static `html` and `js` file which I then import to `electron`.

I would like to encrypt this `.js` file, preferably with some automatic way while building with `electron-builder` or vite's included `rollup`.

I have successfully used `bytenode` for the electron `.js` files but can't figure out how to do the same for the client files.
How would you suggest is the best way to compile and load it within the html?

index.html
```html
<html lang="en">
  <head>
    <link rel="stylesheet" href="/_assets/style.8c7bb816.css">
  </head>
  <body class="w-screen h-full min-h-screen min-w-screen font-body">
    <div id="app"></div>
    <script>
      // I assume 'bytenode' and 'fs' will be loaded by electron's node_modules
      const fs = require('fs');
      const bytenode = require('bytenode');
      if (!fs.existsSync('/_assets/main.jsc')) {
        bytenode.compileFile('/_assets/main.js', '/_assets/main.jsc');
      }

      //How exactly should I load the jsc? This was loaded as a module before...
      require('/_assets/main.jsc'); //<-- Example's getHTML function is not available within vue. Should I just request the jsc and only that?

    </script>
    <!--  <script type="module" src="/_assets/main.js"></script>-->
</body>
</html>
```
Unfortunately the above does not compile automatically when built with electron.

I would greatly appreciate your help.
Many Thanks

**Comment by cliqer** [comment]
@OsamaAbbas it's been a while since I posted this and unfortunately the above doesn't work.
Can you please give me a hint of how to proceed?
It's been killing me for a while so any help would be very much appreciated.
Many Thanks

**Comment by OsamaAbbas** [comment]
I apologize for the late response. I will look into it today and get back to you.

**Comment by OsamaAbbas** [comment]
Can you please make a minimal example that mimic your desired setup and that works with plain `.js` files?

I'm trying to figure out how to protect `main.js` –which is supposed to be included via `<script>` tag–, but without an example it's not easy.

Create a new repo or email me with your example.

**Comment by OsamaAbbas** [comment]
A possible workaround:

use `nodeIntegration: true` when creating your `BrowserWindow` in your `main.js` electron file.

Also, read this [ [Do not enable Node.js Integration for Remote Content](https://github.com/electron/electron/blob/master/docs/tutorial/security.md#2-do-not-enable-nodejs-integration-for-remote-content) ] if your app uses remote content that you don't control.

Please let me know if this solves your issue. I tried a very minimal example that looks like your code above and it worked just fine.

**Comment by cliqer** [comment]
Thank you @OsamaAbbas for having a look on it.
If you are familiar with sandbox you could use my vite example from there.
[https://codesandbox.io/s/vite-tailwind-pug-04mrv](https://codesandbox.io/s/vite-tailwind-pug-04mrv)
Just download and run `vite build` to create the static files.
Hope this helps.

**Comment by OsamaAbbas** [comment]
Have you tried the workaround mentioned just before your comment?

**Comment by OsamaAbbas** [comment]
I have generated the static files on my machine, changed the `index.html` to something like this:

```html
  <script>
    const fs = require('fs');

    const bytenode = require('bytenode');

    if (!fs.existsSync('./_assets/index.f75bf535.jsc')) {
      bytenode.compileFile('./_assets/index.f75bf535.js', './_assets/index.f75bf535.jsc');
    }

    require('./_assets/index.f75bf535.jsc');
  </script>
  <!-- <script src="./_assets/index.f75bf535.jsc"></script> -->
```

Then I added `nodeIntegration: true` as mentioned above:

```javascript
const mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      preload: path.join(__dirname, 'preload.js'),
      nodeIntegration: true
    }
  })
```

It now works. Please try it yourself.

If you have any other question feel free to ask.

**Comment by cliqer** [comment]
@OsamaAbbas yes I tried before I post and no it doesn't compile on my case and being on windows when building electron.
Can you please share the 'preload.js' as for me to understand better how this works?
Many Thanks

**Comment by OsamaAbbas** [comment]
It's the default empty file that comes with electron hello-world example.

I will try to share with you the complete setup tonight.

**Comment by OsamaAbbas** [comment]
I have tested it now on a windows machine. It works.

Have you changed every `/_assets/` reference, in your vite generated code, to `_assets/`? Because otherwise, electron won't find the files.

However, I have just used `vite` to generate the static files, then I put them instead of the static files of the `electron-quick-start` example. I finally run the project with  `electron .`.

If your problem is with `electron-builder`, I guess this needs opening another issue.
